fms/utils/serialization.py

```diff
@@ -405,6 +473,11 @@ def load_state_dict_into_model(
     # 1. Get the adapter from checkpoint sd to fms sd
     adapter = _get_adapter(architecture, source)
 
+    # Prepare the extra_kwargs for the adapter
+    adapter_kwargs = {}
+    if hasattr(model, "config"):
+        adapter_kwargs["model_config"] = model.config.as_dict()
```

Comment:
any reason to not send in model_config as a `ModelConfig` rather than a `dict`?

Reply:
I thought keeping all the conditionals as key checks in a dict would make the code cleaner, instead of mixing `"model_config" in kwargs and get_attr(kwargs["model_config"], "fused_weights")`